### PR TITLE
Drop Entity DAO support

### PR DESCRIPTION
**Pull request by foxycode**
Related to #238

### CODE REVIEW

**Comment by TomasVotruba** [comment]
:+1: Looks great

File: src/Kdyby/Doctrine/DI/OrmExtension.php [hunk 1]
```diff
@@ -417,20 +417,20 @@ protected function processEntityManager($name, array $defaults)
...
 				->setParameters(['entityName'])
 				->setInject(FALSE);
 
 			// interface for models & presenters
-			$builder->addDefinition($this->prefix('daoFactory'))
-				->setClass('Kdyby\Doctrine\EntityDao')
-				->setFactory('@Kdyby\Doctrine\EntityManager::getDao', [new Code\PhpLiteral('$entityName')])
-				->setParameters(['entityName'])
-				->setImplement('Kdyby\Doctrine\EntityDaoFactory')
+			$builder->addDefinition($this->prefix('repositoryFactory'))
```
Comment (fprochazka) [inline src/Kdyby/Doctrine/DI/OrmExtension.php]: this definition can be removed altogether

File: src/Kdyby/Doctrine/DI/OrmExtension.php [hunk 2]
```diff
@@ -417,20 +417,20 @@ protected function processEntityManager($name, array $defaults)
...
+				->setClass($config['defaultRepositoryClassName'])
+				->setImplement('Kdyby\Doctrine\DI\IRepositoryFactory')
+				->setArguments([new Code\PhpLiteral('$entityManager'), new Code\PhpLiteral('$classMetadata')])
+				->setParameters(['Doctrine\ORM\EntityManagerInterface entityManager', 'Doctrine\ORM\Mapping\ClassMetadata classMetadata'])
 				->setInject(FALSE)->setAutowired(TRUE);
 		}
```
Comment (fprochazka) [inline src/Kdyby/Doctrine/DI/OrmExtension.php]: under this line, there is `...defaultRepositoryFactory` service definition, can you please change `->setAutowired(FALSE)` to `->setAutowired($isDefault)` ?

File: src/Kdyby/Doctrine/DI/OrmExtension.php [hunk 3]
```diff
@@ -417,20 +417,20 @@ protected function processEntityManager($name, array $defaults)
 			$entityManager->addSetup('?->bindEntityManager(?)', [$this->prefix('@' . $name . '.diagnosticsPanel'), '@self']);
 		}
 
-		if ($isDefault && $config['defaultRepositoryClassName'] === 'Kdyby\Doctrine\EntityDao') {
+		if ($isDefault && $config['defaultRepositoryClassName'] === 'Kdyby\Doctrine\EntityRepository') {
 			// syntax sugar for config
-			$builder->addDefinition($this->prefix('dao'))
-				->setClass('Kdyby\Doctrine\EntityDao')
-				->setFactory('@Kdyby\Doctrine\EntityManager::getDao', [new Code\PhpLiteral('$entityName')])
+			$builder->addDefinition($this->prefix('repository'))
```
Comment (fprochazka) [inline src/Kdyby/Doctrine/DI/OrmExtension.php]: I'm thinking, this can be removed too

File: src/Kdyby/Doctrine/DI/OrmExtension.php [hunk 4]
```diff
@@ -417,20 +417,20 @@ protected function processEntityManager($name, array $defaults)
 			$entityManager->addSetup('?->bindEntityManager(?)', [$this->prefix('@' . $name . '.diagnosticsPanel'), '@self']);
 		}
 
-		if ($isDefault && $config['defaultRepositoryClassName'] === 'Kdyby\Doctrine\EntityDao') {
+		if ($isDefault && $config['defaultRepositoryClassName'] === 'Kdyby\Doctrine\EntityRepository') {
```
Comment (fprochazka) [inline src/Kdyby/Doctrine/DI/OrmExtension.php]: this whole branch can probably go away.

**Comment by foxycode** [comment]
@fprochazka Cleaned more and tests are still passing, but I think there could be more cleanup.

What can't I understand is, why `RepositoryFactory` definitions in `OrmExtension` are duplicate.
here: https://git.io/vakAT
and here: https://git.io/vakA3

Are they needed both?

**Comment by fprochazka** [comment]
@foxycode 
* one is generated factory to create default repository, if entity has not defined custom repository, to still allow dependency injection for that repository
* the other one is [RepositoryFactory](https://api.kdyby.org/class-Doctrine.ORM.Repository.RepositoryFactory.html) implementation

**Comment by fprochazka** [comment]
Looks good, thank you!

**Pull request merged**